qutebrowser/browser/downloads.py

```diff
@@ -894,110 +894,196 @@ def _on_data_changed(self, idx, *, webengine):
             qtutils.ensure_valid(end_index)
         self.dataChanged.emit(start_index, end_index)
 
-    def _raise_no_download(self, count):
-        """Raise an exception that the download doesn't exist.
+    def _selective_dos(self, count, all_, indexset, criterion, dos,
```

Comment:
I think it's fine to return a list in this case - because (as @Al-Caveman mentioned) at least with the current code, the list of downloads is getting mutated while iterating over the matched downloads.